ion-schema/src/main/kotlin/com/amazon/ionschema/model/TimestampPrecisionValue.kt

```diff
@@ -0,0 +1,41 @@
+package com.amazon.ionschema.model
+
+/**
+ * Represents the available arguments for the `timestamp_precision` constraint.
+ * @see Constraint.TimestampPrecision
+ */
+enum class TimestampPrecisionValue(internal val id: Int) {
```

Comment:
Kotlin internal doesn't provide protection to consumers writing java (at least it didn't in previous versions). I would say just make it private or not (the nice part about kotlin is we can always set an `id` property later if need be).

Reply:
I thought the Kotlin compiler did some name mangling for `internal` items (protection through obscurity), and if so, I would be satisfied with that. However, I can refactor this so that `id` can be private.